### PR TITLE
Add NumerSense

### PR DESCRIPTION
**Pull request by joeddav**
Adds the NumerSense dataset
- Webpage/leaderboard: https://inklab.usc.edu/NumerSense/
- Paper: https://arxiv.org/abs/2005.00683
- Description: NumerSense is a new numerical commonsense reasoning probing task, with a diagnostic dataset consisting of 3,145 masked-word-prediction probes. Basically, it's a benchmark to see whether your MLM can figure out the right number in a fill-in-the-blank task based on commonsense knowledge (a bird has **two** legs)

### CODE REVIEW

File: datasets/numer_sense/README.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+- crowdsourced
+languages:
+- en
+licenses:
+- mit
+multilinguality:
+- monolingual
+size_categories:
+- 10K<n<100K
+source_datasets: []
```
Comment (lhoestq) [inline datasets/numer_sense/README.md:14]: If the dataset use original data please use

```suggestion
source_datasets:
- original
```

File: datasets/numer_sense/README.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+Each value of the training set consists of (1) a sentence with a number masked out with the `<mask>` token, and (2) a
+ground truth target value (both strings). The test sets include the masked sentences but without the ground truth
+values.
+
+### Data Splits
+
+The dataset includes the following pre-defined data splits:
+
+- A train set with >10K labeled examples (i.e. containing a ground truth value)
+- A core test set (`test_core`) with 1,132 examples (no ground truth provided)
```
Comment (lhoestq) [inline datasets/numer_sense/README.md:94]: could you make it explicit that we use the empty string `""` for no label for this split ?

**Review by lhoestq — APPROVED** [review]
LGTM !

Feel free to merge if you have more more changes to do

File: datasets/numer_sense/README.md (hunk 1 above)
Comment (joeddav) [reply datasets/numer_sense/README.md:14]: It doesn't use original data, but the source dataset isn't in the library so @yjernite suggested leaving the tag blank and linking to the source dataset in the card below.

Comment (yjernite) [reply datasets/numer_sense/README.md:14]: Sorry that was a miscommunication: this should be `extended|other`

If you get pull the latest tagger, you can also enter the name for the other

**Pull request merged**